test/tools/cli/IonCompare.java

```diff
@@ -0,0 +1,771 @@
+import tools.errorReport.ErrorDescription;
+import tools.errorReport.ErrorType;
+import tools.events.Event;
+import tools.events.EventType;
+import tools.events.ImportDescriptor;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class IonCompare {
+    private static final int CONSOLE_WIDTH = 120; // Only used for formatting the USAGE message
+    private static final int USAGE_ERROR_EXIT_CODE = 1;
+    private static final int IO_ERROR_EXIT_CODE = 2;
+    private static final int BUFFER_SIZE = 128 * 1024;
+    private static final String SYSTEM_OUT_DEFAULT_VALUE = "out";
+    private static final String SYSTEM_ERR_DEFAULT_VALUE = "err";
+    private static final IonSystem ION_SYSTEM = IonSystemBuilder.standard().build();
+    private static final IonTextWriterBuilder ION_TEXT_WRITER_BUILDER = IonTextWriterBuilder.standard();
+    private static final String EMBEDDED_STREAM_ANNOTATION = "embedded_documents";
+
+    public static void main(final String[] args) {
+        CompareArgs parsedArgs = new CompareArgs();
+        CmdLineParser parser = new CmdLineParser(parsedArgs);
+        parser.getProperties().withUsageWidth(CONSOLE_WIDTH);
+        ComparisonType comparisonType = parsedArgs.getComparisonType();
+
+        try {
+            parser.parseArgument(args);
+            comparisonType = parsedArgs.getComparisonType();
+        } catch (CmdLineException | IllegalArgumentException e) {
+            System.err.println(e.getMessage() + "\n");
+            System.err.println("\"Compare\" Compare all inputs (which may contain Ion streams and/or EventStreams) \n"
+                    + "against all other inputs using the Ion data model's definition of equality. Write a \n"
+                    + "ComparisonReport to the output.\n");
+            System.err.println("Usage:\n");
+            System.err.println("ion compare [--output <file>] [--error-report <file>] [--output-format (text | \n"
+                    + "pretty | binary | none)]  [--catalog <file>]... [--comparison-type (basic | equivs | \n"
+                    + "non-equivs | equiv-timeline)] [-] [<input_file>]...\n");
+            parser.printUsage(System.err);
+            System.exit(USAGE_ERROR_EXIT_CODE);
+        }
+
+        try (//Initialize output stream, never return null. (default value: STDOUT)
+             OutputStream outputStream = initOutputStream(parsedArgs, SYSTEM_OUT_DEFAULT_VALUE);
+             //Initialize error report, never return null. (default value: STDERR)
+             OutputStream errorReportOutputStream = initOutputStream(parsedArgs, SYSTEM_ERR_DEFAULT_VALUE);
+             IonWriter ionWriterForOutput = IonTextWriterBuilder.pretty().build(outputStream);
```

Comment:
Changed.

In case I miss anything, Ion-c uses text format for comparison report no matter what format I used for -f. Do I need to match ion-c's performance here?

Reply:
Yeah it's fine to use text for the comparison report.